src/main/java/org/apache/cassandra/sidecar/metrics/MetricRegistryFactory.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.metrics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SharedMetricRegistries;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.apache.cassandra.sidecar.config.SidecarConfiguration;
+
+/**
+ * Provider for getting {@link FilteringMetricRegistry} based on provided metrics configuration
+ */
+@Singleton
+public class MetricRegistryFactory
+{
+    private final String globalRegistryName;
+    private final ReadWriteLock readWriteLock;
+    private final List<MetricFilter> inclusions;
+    private final List<MetricFilter> exclusions;
+
+    @Inject
+    public MetricRegistryFactory(SidecarConfiguration sidecarConfiguration)
+    {
+        this(sidecarConfiguration.metricsConfiguration().registryName(),
+             MetricFilter.parse(sidecarConfiguration.metricsConfiguration().includeConfigurations()),
+             MetricFilter.parse(sidecarConfiguration.metricsConfiguration().excludeConfigurations()));
+    }
+
+    @VisibleForTesting
+    public MetricRegistryFactory(String globalRegistryName, List<MetricFilter> inclusions, List<MetricFilter> exclusions)
```

Comment:
use package private for testing?
```suggestion
   MetricRegistryFactory(String globalRegistryName, List<MetricFilter> inclusions, List<MetricFilter> exclusions)
```

Reply:
This is useful in general, if people want to pass filters and create the class, instead of always wanting `SidecarConfiguration` for creating a factor.

Reply:
fair, makes sense